### PR TITLE
Replace LibParser with tree-sitter-darklang parser in some test files + some grammar updates

### PR DESCRIPTION
**Pull request by OceanOak**
Changelog:

```
Parser:
- Replace LibParser with tree-sitter-darklang parser in some test files
- Add support for multi-line lists and record literals where a newline is considered a delimiter
```
LibParser was replaced in: 
`BwdServer.Tests`, `Canvas.Tests`, `HttpClient.Tests`, `ProgramTypes.Tests`, `Queue.Tests`, `QueueSchedulingRules.Tests.fs`

### CODE REVIEW

File: backend/testfiles/httphandler/_simple-request-headers.test [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [http-handler GET /]
 (let body =
-   request.headers
+  (request.headers
    |> PACKAGE.Darklang.Stdlib.List.map (fun (k,v) -> "(\"" ++ k ++ "\", \"" ++ v ++ "\")")
```
Comment (OceanOak) [inline backend/testfiles/httphandler/_simple-request-headers.test:4]: TODO: Bring this test back (having issues parsing `(\""`)

File: backend/testfiles/httphandler/injected-icon-post-roundtrip.test [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [http-handler POST /]
-PACKAGE.Darklang.Stdlib.Http.response request.body 200L
+PACKAGE.Darklang.Stdlib.Http.response (request.body) 200L
```
Comment (OceanOak) [inline backend/testfiles/httphandler/injected-icon-post-roundtrip.test:2]: TODO: in the grammar consider moving `field_access` to `simple_expression` so we don't have to wrap this in parens

File: backend/testfiles/httphandler/_simple-request-headers.test (hunk 1 above)
Comment (StachuDotNet) [reply backend/testfiles/httphandler/_simple-request-headers.test:4]: maybe inline this TODO/CLEANUP in the file somewhere?

File: packages/darklang/languageTools/parser/parserTest.dark [hunk 3]
```diff
@@ -8,6 +8,16 @@ module Darklang =
...
 
+        let parseWTExpr (code: WrittenTypes.ParsedFile) : WrittenTypes.Expr =
+          let exprs =
+            match sourceFile with
+            | SourceFile c -> c.exprsToEval
+            | _ -> []
+
+          match exprs with
+          | [ expr ] -> expr
+          | _ -> failwith "Expected exactly one expression to be parsed"
```
[the remark this answers is not in the document]
Comment (OceanOak) [reply packages/darklang/languageTools/parser/parserTest.dark]: Oops, probably copilot, sorry. Fixed

[the remark this answers is not in the document]
Comment (OceanOak) [reply packages/darklang/languageTools/parser/parserTest.dark]: should it return a Result instead?

[the remark this answers is not in the document]
Comment (StachuDotNet) [reply packages/darklang/languageTools/parser/parserTest.dark]: Yeah, it probably should

[the remark this answers is not in the document]
Comment (OceanOak) [reply packages/darklang/languageTools/parser/parserTest.dark]: The function wasn't used; I had it when I was sketching a solution

**Pull request merged**